server/lib/nl/fulfillment/filter_with_dual_vars.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import logging
+from typing import List
+
+from server.lib.nl import utils
+from server.lib.nl.detection import ContainedInPlaceType
+from server.lib.nl.detection import Place
+from server.lib.nl.fulfillment.base import add_chart_to_utterance
+from server.lib.nl.fulfillment.base import ChartVars
+from server.lib.nl.fulfillment.base import open_top_topics_ordered
+from server.lib.nl.fulfillment.base import populate_charts
+from server.lib.nl.fulfillment.base import PopulateState
+from server.lib.nl.utterance import ChartOriginType
+from server.lib.nl.utterance import ChartType
+from server.lib.nl.utterance import Utterance
+
+# TODO: Support per-capita
+# TODO: Increase this after frontend handles comparedPlaces better.
+_MAX_PLACES_TO_RETURN = 7
+
+
+def _get_dual_sv_part(uttr: Utterance, idx: int) -> List[str]:
```

Comment:
nit: since this is picking the dual part and that's always the second one, why not fix the idx = 1 here instead of relying on the caller to specify it? If the caller specifies idx = 0, then that means this won't be a "dual" part extractor?

Reply:
This function needed comments (added now). It is not always picking the 2nd one, rather it finds the dual-sv candidate (from the multi-sv results), and for that candidate returns either the first or second part, called from different places.